### PR TITLE
Remove two problematic characters from Routing Transparency test

### PR DESCRIPTION
**Pull request by dimivel**
### Are you submitting this PR against the [develop branch](https://github.com/cloudfoundry/cf-acceptance-tests/tree/develop)?

_All PR's to CATs should be submitted to develop and will be merged to main once they've passed acceptance._

### What is this change about?

Remove the two problematic characters from Routing Transparency test that should appropriately handles certain reserved/unsafe characters but its not the case.

### Please provide contextual information.

_Include any links to other PRs, stories, slack discussions, etc... that will help establish context._
https://github.com/cloudfoundry/cf-acceptance-tests/issues/1276

### What version of cf-deployment have you run this cf-acceptance-test change against?

v44.0.0

### Please check all that apply for this PR:

- [ ] introduces a new test --- Are you sure everyone should be running this test?
- [x] changes an existing test
- [ ] requires an update to a CATs integration-config

### Did you update the README as appropriate for this change?

- [ ] YES
- [x] N/A

### If you are introducing a new acceptance test, what is your rationale for including it CATs rather than your own acceptance test suite?

_CATs should validate common operator workflows._
_CATs is not a regression test suite._
_CATs is run by every component team to validate their releases before promotion._

### What is the level of urgency for publishing this change?

- [ ] **Urgent** - unblocks current or future work
- [ ] **Slightly Less than Urgent**

### Tag your pair, your PM, and/or team!
_It's helpful to tag a few other folks on your team or your team alias in case we need to follow up later._

### CODE REVIEW

File: apps/routing_transparency.go [hunk 1]
```diff
@@ -44,11 +44,11 @@ var _ = AppsDescribe("Routing Transparency", func() {
 	})
 
 	It("appropriately handles certain reserved/unsafe characters", func() {
-		curlResponse := helpers.CurlApp(Config, appName, "/requesturi/!~^'()$\"?!'()$#!'")
```
Comment (jochenehret) [inline apps/routing_transparency.go:47]: Better solution: Check if `GetIncludeHTTP2Routing()` is `false` and only then run the double-quote test.

Comment (dimivel) [reply apps/routing_transparency.go]: Done, please review if our intention is correct.

Comment (jochenehret) [reply apps/routing_transparency.go]: That would skip the whole test if HTTP/2 is enabled. What I meant is just skipping the " test and only enable it if HTTP/2 is disabled. Something like this (not tested):
```
	It("appropriately handles certain reserved/unsafe characters", func() {
		curlResponse := helpers.CurlApp(Config, appName, "/requesturi/!~^'()$?!'()$#!'") // no " in request uri
		Expect(curlResponse).To(ContainSubstring("Request"))

		By("preserving all characters")
		Expect(curlResponse).To(ContainSubstring("/requesturi/!~^'()$"))
		Expect(curlResponse).To(ContainSubstring("Query String is [!'()$]"))

		By("preserving double quotes (not for HTTP/2)")
		if !Config.GetIncludeHTTP2Routing() {
			curlResponse = helpers.CurlApp(Config, appName, "/requesturi/\"")
			Expect(curlResponse).To(ContainSubstring("/requesturi/\""))
		}
	})
```

Comment (dimivel) [reply apps/routing_transparency.go:47]: The proposal above is tested locally and works as expected.

**Review by jochenehret — APPROVED** [review]

**Review by ctlong — APPROVED** [review]

**Pull request merged**